src/sentry/api/helpers/group_index/update.py

```diff
@@ -738,3 +715,45 @@ def update_groups(
         result["inbox"] = inbox
 
     return Response(result)
+
+
+def handle_is_public(
+    is_public: Any,
+    group_list: List[Group],
+    project_lookup: Dict[int, Project],
+    acting_user: User | None,
+) -> str | None:
+    """
+    Handle the isPublic flag on a group update.
+
+    This deletes the existing share ID and creates a new share ID if isPublic is True.
+    We always want to delete an existing share, because triggering an isPublic=True
+    even when it's already public, should trigger regenerating.
```

Comment:
```suggestion
    This deletes the existing share ID and creates a new share ID if isPublic is True.
    We always want to delete an existing share, because triggering an isPublic=True
    when it's already public should trigger regenerating.
```